tests/src/tests/test_utils.go

```diff
@@ -0,0 +1,394 @@
+			testLogger = log.New(f, "", log.LstdFlags|log.Lmicroseconds)
+		} else {
+			writer := io.MultiWriter(f, os.Stdout)
+			testLogger = log.New(writer, "", log.LstdFlags|log.Lmicroseconds)
+		}
+		testLogger.Println("Test Starting:")
+	}
+}
+
+// Called from a test program.
+//  - determines the test program name
+// 	- creates a temproray directory for the test program
+// returns the name of the directory created.
+func GetTempDir() string {
+	_, fn, _, ok := runtime.Caller(1)
+	if !ok {
+		return tmpDir
+	}
+	testFile := filepath.Base(fn)
+	testFileExtension := filepath.Ext(testFile)
+	subdir := testFile[0 : len(testFile)-len(testFileExtension)]
+	return CreateTempDir(subdir)
+}
+
+// Creates a specified sub directory under the temp directory if it does not exists
+// Returns the name of the temp directory.
+func CreateTempDir(subdir string) string {
+	tempDir := tmpDir + subdir + "/"
+	if _, err := os.Stat(tempDir); os.IsNotExist(err) {
+		os.Mkdir(tempDir, 0755)
+	}
+	return tempDir
+}
+
+// Called from a test program.
+//   - ensure the temp directory for the test program exists
+//   - generates a devfile name badsed in the calling function name
+// Returns the devfile name
+func GetDevFileName() string {
+	pc, fn, _, ok := runtime.Caller(1)
+	if !ok {
+		return tmpDir + "DefaultDevfile"
+	}
+
+	testFile := filepath.Base(fn)
+	testFileExtension := filepath.Ext(testFile)
+	subdir := testFile[0 : len(testFile)-len(testFileExtension)]
+	destDir := CreateTempDir(subdir)
+	callerName := runtime.FuncForPC(pc).Name()
+
+	pos1 := strings.LastIndex(callerName, "/tests.") + len("/tests.")
+	devfileName := destDir + callerName[pos1:len(callerName)] + ".yaml"
+
+	LogInfoMessage(fmt.Sprintf("GetDevFileName : %s", devfileName))
+
+	return devfileName
+}
+
+// Adds a specified suffix to the name of a specified file.
+// For example if the file is devfile.yaml and the suffix 1 the result is devfile1.yaml
+func AddSuffixToFileName(fileName string, suffix string) string {
+	pos1 := strings.LastIndex(fileName, ".yaml")
+	newFileName := fileName[0:pos1] + suffix + ".yaml"
+	LogInfoMessage(fmt.Sprintf("Add suffix %s to fileName %s : %s", suffix, fileName, newFileName))
+	return newFileName
+}
+
+// Log the specified message
+// Return the message logged
+func LogMessage(message string) string {
+	testLogger.Println(message)
+	return message
+}
+
+// Log the specified message as an Error
+// Return the message logged
+var errorPrefix = "..... ERROR : "
+
+func LogErrorMessage(message string) string {
+	var errMessage []string
+	errMessage = append(errMessage, errorPrefix, message)
+	return LogMessage(fmt.Sprint(errMessage))
+}
+
+// Log the specified message as Info
+// Return the message logged
+var infoPrefix = "INFO :"
+
+func LogInfoMessage(message string) string {
+	var infoMessage []string
+	infoMessage = append(infoMessage, infoPrefix, message)
+	return LogMessage(fmt.Sprint(infoMessage))
+}
+
+type TestDevfile struct {
+	SchemaDevFile   schema.Devfile
```

Comment:
can we rename this to something else? Its very confusing to read `TestDevfile.SchemaDevFile`. Maybe Data or Contents sounds ok?

Reply:
I honestly cannot think of a more descriptive name - going to punt in this one.